model_evaluation/utils_export/dataset_test.py

```diff
@@ -91,55 +91,92 @@ def testBadTypeFeatureKeys(self):
-        model_names=None,
-        project_name=None
-        )
+        model_names='None',
+        project_name=None)
 
     def input_fn(max_n_examples):
-      return pd.DataFrame(
-          {'other_feature': ['This is one'] * max_n_examples,
-           'label_name': [0] * max_n_examples})
+      return pd.DataFrame({
+          'other_feature': ['This is one'] * max_n_examples,
+          'label_name': [0] * max_n_examples
+      })
 
     with self.assertRaises(Exception) as context:
-      dataset = Dataset(input_fn)
+      dataset = Dataset(input_fn, 'dataset_dir')
       dataset.check_compatibility(model)
-      self.assertIn(
-          'input_fn must contain at least the feature keys',
-          str(context.exception))
+      self.assertIn('input_fn must contain at least the feature keys',
+                    str(context.exception))
 
   def testModelIsCompatibleWithDataset(self):
     model = Model(
         feature_keys_spec={'comment_text': EncodingFeatureSpec.LIST_STRING},
         prediction_keys='prediction_key',
-        model_names=None,
-        project_name=None
-        )
+        model_names='None',
+        project_name=None)
 
     def input_fn(max_n_examples):
-      return pd.DataFrame(
-          {'comment_text': ['This is one'] * max_n_examples,
-           'label_name': [0] * max_n_examples})
+      return pd.DataFrame({
+          'comment_text': ['This is one'] * max_n_examples,
+          'label_name': [0] * max_n_examples
+      })
 
     try:
-      dataset = Dataset(input_fn)
+      dataset = Dataset(input_fn, 'dataset_dir')
       dataset.check_compatibility(model)
     except ValueError:
       self.fail('Dataset raised an exception unexpectedly!')
 
 
+class TestEndPipeline(unittest.TestCase):
+  """Verifies end-to-end use of dataset."""
+
+  def setUp(self):
+    def input_fn_test(max_n_examples):
+      return pd.DataFrame({
+          'comment_text': [['This', 'is', 'one']] * max_n_examples
+      })
+
+    gcs_path_test = os.path.join('gs://kaggle-model-experiments/',
+                                 getpass.getuser(), 'unittest')
+    self.dataset = Dataset(input_fn_test, gcs_path_test)
+    self.dataset.load_data(5)
+
+    model_input_spec = {
+        'comment_text': EncodingFeatureSpec.LIST_STRING,
+    }
+    self.model = Model(
+        feature_keys_spec=model_input_spec,
+        prediction_keys='frac_neg/logistic',
+        example_key='comment_key',
+        model_names=[
+            'tf_gru_attention:v_20180914_163804',
+            'tf_gru_attention:v_20180823_133625'
+        ],
+        project_name='wikidetox')
+
+  def testComputePredictions(self):
+    try:
+      self.dataset.add_model_prediction_to_data(self.model) 
+    except ValueError :
+      self.fail('Dataset raised an exception unexpectedly!')
+
+  def testLoadPredictions(self):
+    try:
+      self.dataset.add_model_prediction_to_data(
+          self.model, recompute_predictions=False)
```

Comment:
Is there an assert we can use to check that the predictions were loaded and not recomputed?

Reply:
Would that make sense to delete test directory before? It would ensure that predictions are not loaded.
However, There is the risk that this directory contains valuable information (limited risk if the test directory has a "unusual enough" name)

Reply:
Modified by adding a version.